Minor update: (1) added hrrrdas ensemble file dependencies to the gsi…

EdwardColon-NOAA opened the pull request:
… step and a series of minor bug fixes in the conus and alaska rhist exscripts.

EdwardColon-NOAA committed:
Minor update: (1) added hrrrdas ensemble file dependencies to the gsi step and a series of minor bug fixes in the conus and alaska rhist exscripts.

MatthewMorris-NOAA commented:
Hi @EdwardColon-NOAA, Is the data dependency age for 10 minutes or 10 seconds? Also, we need a way to run the analysis using GDAS files as back-up in the event the HRRRDAS files are significantly delayed or missing.  Can we add in a hard-wired start time, where the analysis will run at a given time, say 32 minutes, even if the HRRRDAS dependencies are not met?

EdwardColon-NOAA commented:
Hi Matt,

The age dependency is given in seconds assuming that a partially written
file would be completed within that time frame. I'll add an "or" dependency
for time limit that will trigger the use of gdas but let's see if we can
get this added first at a minimum.

-Edward

On Wed, Apr 16, 2025 at 3:17 PM MatthewMorris-NOAA ***@***.***>
wrote:

> Hi @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA>, Is the data
> dependency age for 10 minutes or 10 seconds? Also, we need a way to run the
> analysis using GDAS files as back-up in the event the HRRRDAS files are
> significantly delayed or missing. Can we add in a hard-wired start time,
> where the analysis will run at a given time, say 32 minutes, even if the
> HRRRDAS dependencies are not met?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/41#issuecomment-2810524503>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQF2NH6QJQWJNPTK6DD2Z2UCPAVCNFSM6AAAAAB3JCCDRKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMJQGUZDINJQGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> *MatthewMorris-NOAA* left a comment (NOAA-EMC/NOAA_3drtma#41)
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/41#issuecomment-2810524503>
>
> Hi @EdwardColon-NOAA <https://github.com/EdwardColon-NOAA>, Is the data
> dependency age for 10 minutes or 10 seconds? Also, we need a way to run the
> analysis using GDAS files as back-up in the event the HRRRDAS files are
> significantly delayed or missing. Can we add in a hard-wired start time,
> where the analysis will run at a given time, say 32 minutes, even if the
> HRRRDAS dependencies are not met?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/41#issuecomment-2810524503>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQF2NH6QJQWJNPTK6DD2Z2UCPAVCNFSM6AAAAAB3JCCDRKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQMJQGUZDINJQGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
-------------------------------------------------
Edward Colón
*Lynker* at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

MatthewMorris-NOAA commented:
Hi Ed, I'd prefer to see the full dependency in place before this PR is merged and used in the official parallels.

EdwardColon-NOAA committed:
Update rtma3d_pbspro_conus.xml

Added a HRRRDAS time cutoff.

MatthewMorris-NOAA commented on workflow/rtma3d_pbspro_conus.xml:
```diff
@@ -1170,9 +1176,48 @@
+           <datadep age="10"><cyclestr offset="-3600">&COMINHRRRDAS;/hrrrdas_small_d02_@Y@m@d@H00f01_mem0035</cyclestr></datadep>
+           <datadep age="10"><cyclestr offset="-3600">&COMINHRRRDAS;/hrrrdas_small_d02_@Y@m@d@H00f01_mem0036</cyclestr></datadep>
+           <or>
+		<timedep><cyclestr @Y@m@d@H&HRRRDAS_CUTOFF;@S</cyclestr></timedep>
```
There is an error in this timdep line.

MatthewMorris-NOAA reviewed (changes requested):
Looks good, overall, but there was a typo in the timedep line.

EdwardColon-NOAA commented:
Thanks,

I'll fix it and am adding the logic to the gsianl exscript.

-Edward

On Wed, Apr 16, 2025 at 3:57 PM MatthewMorris-NOAA ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Looks good, overall, but there was a typo in the timedep line.
> ------------------------------
>
> In workflow/rtma3d_pbspro_conus.xml
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/41#discussion_r2047648361>:
>
> > +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0023</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0024</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0025</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0026</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0027</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0028</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0029</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0030</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0031</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0032</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0033</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0034</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0035</cyclestr></datadep>
> +           <datadep age="10"><cyclestr ***@***.***@***@***.***@H00f01_mem0036</cyclestr></datadep>
> +           <or>
> +		<timedep><cyclestr @***@***.***@***@***.******@***.***</cyclestr></timedep>
>
> There is an error in this timdep line.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/NOAA_3drtma/pull/41#pullrequestreview-2773686203>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACFXXQG7KI6PTZFSFWMDNXL2Z2YY5AVCNFSM6AAAAAB3JCCDRKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDONZTGY4DMMRQGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
-------------------------------------------------
Edward Colón
*Lynker* at NOAA/NWS/NCEP/EMC
5830 University Research Ct.
Rm #2025
Riverdale, MD 20737
Office: (301) 683-3815
Cell: (301) 213-3566

EdwardColon-NOAA committed:
Update exrtma3d_gsianl.ksh

Added a conditional statement that checks if the gsi start time is less than the HRRRDAS cutoff. If it is, the analysis will proceed with the use of HRRRDAS ensemble files. If not then gdas ensemble files will be used.

EdwardColon-NOAA committed:
Update rtma3d_pbspro_conus.xml

MatthewMorris-NOAA commented on workflow/rtma3d_pbspro_conus.xml:
```diff
@@ -1216,7 +1216,7 @@
            <datadep age="10"><cyclestr offset="-3600">&COMINHRRRDAS;/hrrrdas_small_d02_@Y@m@d@H00f01_mem0036</cyclestr></datadep>
            <or>
-		<timedep><cyclestr @Y@m@d@H&HRRRDAS_CUTOFF;@S</cyclestr></timedep>
+		<timedep><cyclestr>@Y@m@d@H&HRRRDAS_CUTOFF;@S</cyclestr></timedep>
```
There is still the issue of HRRRDAS_CUTOFF being in the middle of the time string.

EdwardColon-NOAA committed:
Update rtma3d_pbspro_conus.xml

Introduced offset into time dependency.

MatthewMorris-NOAA commented on workflow/rtma3d_pbspro_conus.xml:
```diff
@@ -200,7 +202,7 @@
 <!ENTITY ACCOUNT         "RTMA-DEV">
 <!ENTITY QUEUE           "dev">
-<!ENTITY QUEUE_DBG       "debug">
+<!ENTITY QUEUE_DBG       "debug">/time_int\
```
what is the reason for adding time_int to this?

MatthewMorris-NOAA reviewed (commented):
The way the dependency is written now will force the GSI step to run even if the prepobs and prepfgs steps haven't run.  Is that the intention?

I would have thought that we would require the prepobs and prepfgs steps to run. Then have an either or dependency where either the 36 HRRRDAS members are available or we run at 32 past with GDAS instead.

EdwardColon-NOAA committed:
Update rtma3d_pbspro_conus.xml

EdwardColon-NOAA committed:
Update rtma3d_pbspro_conus.xml

Fixed the dependency nesting.
